pkg/cmd/server/server.go

```diff
@@ -125,13 +126,21 @@ func (s *server) run() error {
 		return err
 	}
 
-	config, err := s.loadConfig()
+	originalConfig, err := s.loadConfig()
 	if err != nil {
 		return err
 	}
+
+	// watchConfig needs to examine the unmodified original config, so we keep that around as a
+	// separate object, and instead apply defaults to a clone.
+	copy, err := scheme.Scheme.DeepCopy(originalConfig)
+	if err != nil {
+		return err
+	}
+	config := copy.(*api.Config)
 	applyConfigDefaults(config)
 
-	s.watchConfig(config)
+	s.watchConfig(originalConfig)
```

Comment:
maybe move this above line 140 so we know at that point we are done with originalConfig 'cos its been handed off to watchConfig.